staging/src/k8s.io/apiserver/pkg/authorization/authorizer/interfaces.go

```diff
@@ -62,6 +65,14 @@ type Attributes interface {
 
 	// GetPath returns the path of the request
 	GetPath() string
+
+	// ParseFieldSelector is lazy, thread-safe, and stores the parsed result and error.
+	// It returns an error if the field selector cannot be parsed.
+	ParseFieldSelector() (fields.Selector, error)
```

Comment:
are we returning the selector interface or just the requirements slice? I expected the latter.

It also looks like the `Requirements()` impl allocs a new array every time, which I'm not sure we want (though we should document the returned object must not be modified (just like the data returned from GetUser().GetGroups() or GetUser().GetExtra())

Reply:
> are we returning the selector interface or just the requirements slice? I expected the latter.

Something broke in the initial factoring.  I can see if I resolved that in the time since.

Reply:
Switching to `[]Requirements` instead of selector will require combining the PRs to have a serializeable representation of the requirements.  Currently `Selector.String()` is used in the caching paths and it'll need a different serializable representation. the `authorizationv1` is a natural one to use.  So I think to satisfy this comment, we'll need to combine this PR and https://github.com/kubernetes/kubernetes/pull/125371